### DISCUSSION TITLE
Installing quarto-cli via pip

### DISCUSSION THREAD

**Discussion by memeplex** [quarto-dev/quarto-cli#8597] [Q&A] [answered]
### Description

For some reason I ended up installing quarto with `pip install quarto-cli`. All the time I thought I had just followed the official installation instructions, but now I come to realize the it's not a documented method. Nevertheless the maintainer is @dragonstyle so I assume it should be ok, shouldn't it? I find the method more convenient than going through the macOS installer or even the brew cask. Also, the VSCode extension  has an option to use the quarto executable that is in the current venv, so at least that's kind of an official acknowledgment of it.

BTW, at the end of https://pypi.org/project/quarto-cli/1.4.549/ it says:

> To uninstall the quarto-cli package, use:
> `pip uninstall jupyterlab_myst`

**Comment by mcanouil** [comment]
Sorry but what's the question? (You put this as a Q&A).

Note that it is not documented because it's a work in progress.
The repository: <https://github.com/quarto-dev/quarto-cli-pypi>

**Reply by memeplex** [reply]
> I assume it should be ok, shouldn't it?

Mainly to confirm it's from you and that I can expect it to be updated regularly, etc.

**Reply by memeplex** [reply]
Moreover, thanks for the link, I'm going to report the little error at the end of https://pypi.org/project/quarto-cli/ there.

**Reply by mcanouil** [reply, marked as answer]
> > I assume it should be ok, shouldn't it?
> 
> 
> 
> Mainly to confirm it's from you and that I can expect it to be updated regularly, etc.
> 
> 

It's a work in progress, so until it is announced and is documented as a download method, using it, is kind of at your own risks.

**Reply by DrOncogene** [reply]
I see it's yet to be documented. What's the update on it?

**Reply by cderv** [reply]
The package is published on pypi for each stable release. So 1.6 latest version is now and it can be used. 
We will keep publishing any patch version for 1.6 and future version, so it can be used. 

It should work as expected. Otherwise, you can report an issue there.

**Comment by ijlyttle** [comment]
Just to add my two cents, being able to install quarto using pip is a game-changer for me:

  - Colleagues can install using tools they are already using:
  
  ```
  pipx install quarto-cli
  ```
  
 - I can install quarto on a GHES Actions runner *without* needing sudo (they don't give us sudo).

I hope this can become part of the "standard" set of tools!

Thanks!

**Comment by nitish1295** [comment]
@cderv  Just checking why isn't this mentioned at https://quarto.org/docs/get-started/. Is it just because it is work in progress? it took me a while to find this discussion (which is a lifesaver)

Installation via pip is very useful, if a build is being released alongside every stable version (and if the build is not an in-progress item/completely unstable item from the maintainers POV) then maybe this should be mentioned in the docs

**Reply by mcanouil** [reply]
I believe the repository is stable now and has patches and new releases synced with Quarto CLI.
You can see that the status on PyPi is "5 - Production/Stable".

The repository for the PyPi build is at https://github.com/quarto-dev/quarto-cli-pypi

Note that this thread is almost two years old.

> Just checking why isn't this mentioned at https://quarto.org/docs/get-started/.

What is "this" you are mentioning?

**Reply by nitish1295** [reply]
Got it, pretty old thread so maybe it was missed. By "this" I mean the method of installation via pip is not mentioned in the installation/get started section at https://quarto.org/docs/get-started/. Am I missing something?  

And it took a really long time to find this discussion and in many cases people may not even think that installation via pip is possible since it was never mentioned on the site. 

Is there a difference is in what we get via pip vs what we get via downloading manually from the get started page (I hope not)

**Reply by mcanouil** [reply]
Unfortunately, it's a bit more general than PIP, there are no installation guide or a comprehensive list of official/community providers (Chocolatey, Conda, PyPi, etc.)

- https://github.com/quarto-dev/quarto-cli/issues/4505

**Reply by nitish1295** [reply]
Got it, thanks for your help!
